src/hooks/useTypesetMath.ts

```diff
@@ -0,0 +1,4 @@
+import React from "react";
+
+export const TypesetMathContext = React.createContext<{ typesetMath?: () => void }>({});
+export const useTypesetMath = () => React.useContext(TypesetMathContext);
```

Comment:
if you initialize to a placeholder you can get rid of your null checks in the usage:

```suggestion
export const TypesetMathContext = React.createContext<() => void>(() => { throw new Error('context not initialized'))});
export const useTypesetMath = () => React.useContext(TypesetMathContext);
```

Reply:
I also like that this forces thinking about handling math content if `Content` gets used outside of the Exercise context.